e2e/epinio.e2e.spec.ts

```diff
@@ -0,0 +1,239 @@
+
+  test.beforeAll(async() => {
+    installEpinioCli();
+    createDefaultSettings();
+    setUpHelmCustomEnv();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  /**
+   * helm teardown
+   * It should run outside of the electronApp.close(), just to make sure the teardown won't
+   * affect the shutdown process in case of exceptions/errors.
+   */
+  test.afterAll(tearDownHelm);
+
+  /**
+   * epinio teardown
+   * It removes epinio binary temp folder along with its binary and
+   * perform a helm uninstall epinio-installer.
+   */
+  test.afterAll(tearDownEpinio);
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: playwrightReportAssets(path.basename(__filename)) });
+    await electronApp.close();
+  });
+
+  test('should start loading the background services', async() => {
+    const navPage = new NavPage(page);
+
+    await navPage.progressBecomesReady();
+    await expect(navPage.progressBar).toBeHidden();
+  });
+  test('should check kubernetes API is ready', async() => {
+    const output = await kubectl('cluster-info');
+
+    expect(output).toMatch(/is running at ./);
+  });
+  test('should verify epinio cli was properly installed', async() => {
+    const epinioCliStatus = await epinio('version');
+
+    expect(epinioCliStatus).toContain('Epinio Version');
+  });
+  test('should add epinio-installer helm repository', async() => {
+    const epinioRepoAdd = await helm('repo', 'add', 'epinio', 'https://epinio.github.io/helm-charts');
+
+    expect(epinioRepoAdd).toContain('"epinio" has been added to your repositories');
+  });
+  test('should install epinio-installer application', async() => {
+    const loadBalancerIpAddr = await loadBalancerIp();
+    const epinioInstall = await helm('install', 'epinio-installer', 'epinio/epinio-installer',
+      '--set', 'skipTraefik=true', '--set', `domain=${ loadBalancerIpAddr }.omg.howdoi.website`,
+      '--wait', '--timeout=25m');
+
+    expect(epinioInstall).toContain('STATUS: deployed');
+  });
+  test('should update epinio config certs file', async() => {
+    const epinioConfigUpdate = await epinio('config', 'update');
+
+    expect(epinioConfigUpdate).toContain('Ok');
+  });
+  test('should push a sample app through epinio cli', async() => {
+    const epinioPush = await epinio('push', '--name', 'sample', '--path', path.join(__dirname, 'assets', 'sample-app'));
+
+    expect(epinioPush).toContain('App is online.');
+  });
+  test('should verify deployed sample application is reachable', async() => {
+    const loadBalancerIpAddr = await loadBalancerIp();
+    const urlAddr = `https://sample.${ loadBalancerIpAddr }.omg.howdoi.website`;
```

Comment:
Do we want to use omg.howdoi.website here? I thought that was mostly for internal use.

Reply:
Well, it's like our own `example.com`. But where are the credentials that let the test runner install stuff on that server? I haven't found any yet.

Reply:
There are no credentials — it functions like `xip.io`. Hence I'm not sure we should use that publicly.

Reply:
Then what should we use?

Reply:
Andrew's using `sslip.io`, which looks more professional and accomplishes the same task, but has the disadvantage of relying on someone else's DNS reflector.